### PR TITLE
Add JDK 17 to the http4s build

### PR DESCRIPTION
**Pull request by vasilmkd**
Use the typelevel jdk index (https://github.com/typelevel/jdk-index).

### CODE REVIEW

File: core/src/main/scala/org/http4s/sbt/Http4sOrgPlugin.scala [hunk 1]
```diff
@@ -42,7 +42,8 @@ object Http4sOrgPlugin extends AutoPlugin {
   val githubActionsSettings: Seq[Setting[_]] =
     Seq(
       spiewakMainBranches := Seq("main"),
-      githubWorkflowJavaVersions := List("adopt@1.8", "adopt@1.11", "adopt@1.16"),
+      githubWorkflowJavaVersions := List("adoptium@8", "adoptium@11", "adoptium@17"),
+      githubWorkflowEnv += ("JABBA_INDEX" -> "https://github.com/typelevel/jdk-index/raw/main/index.json"),
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/sbt/Http4sOrgPlugin.scala:46]: Ah, this is the step that made me worry about transitivity.  I bet this will work.

**Review by armanbilge — APPROVED** [review]

**Review by rossabaker — COMMENTED** [review]
Does this have any CI performance ramifications, downloading a JDK that might not be baked into a GitHub image?

**Comment by rossabaker** [comment]
Well, that CI run didn't look bad at all.  Probably not?

**Comment by vasilmkd** [comment]
@rossabaker `olafurpg/setup-scala` has been downloading JDKs since its first release. 😆 

There is nothing magical done here, jabba the JDK installation tool needs an index file. The original index can be found here https://github.com/shyiko/jabba/blob/cfc25cef4ec67249e17d656d76bd1153bed839cd/index.json, we're simply providing an alternative. The installation then proceeds identically.

**Comment by vasilmkd** [comment]
Technically, none of the JDKs installed using `olafurpg/setup-scala` are baked into the GitHub image.

**Pull request merged**